src/main/scala/utils/MatrixUtils.scala

```diff
@@ -18,9 +18,9 @@ object MatrixUtils extends Serializable {
    * @param mat Input matrix.
    * @return Array of rows.
    */
-  def matrixToRowArray[T](mat: DenseMatrix[T]): Array[DenseVector[T]] = {
+  def matrixToRowArray[T : ClassTag](mat: DenseMatrix[T]): Array[DenseVector[T]] = {
     val matT = mat.t
-    (0 until mat.rows).toArray.map(matT(::, _))
+    (0 until mat.rows).toArray.map(x => DenseVector(matT(::, x).toArray))
```

Comment:
I'm unsure whether .copy and .toDenseVector do the same thing. But I'll add a comment to make it clear why we're doing this.